pkg/agent/manager/manager_test.go

```diff
@@ -1,160 +1,883 @@
+)
+
+func TestShutdownDoesntHangAfterFailedStart(t *testing.T) {
+	trustDomain := "somedomain.com"
+	ca, cakey := createCA(t, trustDomain)
+	baseSVID, baseSVIDKey := createSVID(t, ca, cakey, "spiffe://"+trustDomain+"/agent", 1*time.Hour)
+
+	c := &Config{
+		ServerAddr:  &net.TCPAddr{},
+		SVID:        baseSVID,
+		SVIDKey:     baseSVIDKey,
+		Log:         testLogger,
+		TrustDomain: url.URL{Host: trustDomain},
+	}
+	m, err := New(c)
+	if err != nil {
+		t.Error(err)
+		return
+
+	}
+
+	err = m.Start()
+	if err == nil {
+		t.Errorf("wanted error")
+		return
+	}
+
+	util.RunWithTimeout(t, 1*time.Second, func() {
+		m.Shutdown()
+	})
+}
+
+func TestStoreSVIDOnStartup(t *testing.T) {
+	dir := createTempDir(t)
+	defer removeTempDir(dir)
+
+	trustDomain := "somedomain.com"
+	ca, cakey := createCA(t, trustDomain)
+	baseSVID, baseSVIDKey := createSVID(t, ca, cakey, "spiffe://"+trustDomain+"/agent", 1*time.Hour)
+
+	c := &Config{
+		ServerAddr:    &net.TCPAddr{},
+		SVID:          baseSVID,
+		SVIDKey:       baseSVIDKey,
+		Log:           testLogger,
+		TrustDomain:   url.URL{Host: "somedomain.com"},
+		SVIDCachePath: path.Join(dir, "svid.der"),
+	}
+
+	_, err := ReadSVID(c.SVIDCachePath)
+	if err != ErrNotCached {
+		t.Errorf("wanted: %v, got: %v", ErrNotCached, err)
+		return
+	}
+
+	m, err := New(c)
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	err = m.Start()
+	if err != nil {
+		// Althought start failed, the SVID should have been saved, because it should be
+		// the first thing the manager does at startup.
+		cert, err := ReadSVID(c.SVIDCachePath)
+		if err != nil {
+			t.Error(err)
+			return
+		}
+		if !cert.Equal(baseSVID) {
+			t.Error("SVID was not correctly stored.")
+			return
+		}
+	}
+
+	m.Shutdown()
+	os.Remove(c.SVIDCachePath)
+}
+
+func TestHappyPathWithoutSyncNorRotation(t *testing.T) {
+	dir := createTempDir(t)
+	defer removeTempDir(dir)
+
+	trustDomain := "example.org"
+
+	apiHandler := newMockNodeAPIHandler(&mockNodeAPIHandlerConfig{
+		t:                 t,
+		trustDomain:       trustDomain,
+		dir:               dir,
+		fetchSVIDResponse: fetchSVIDResponseForTestHappyPathWithoutSyncNorRotation,
+		svidTTL:           200,
+	})
+	apiHandler.start()
+	defer apiHandler.stop()
+
+	baseSVID, baseSVIDKey := apiHandler.newSVID("spiffe://"+trustDomain+"/spire/agent/join_token/abcd", 1*time.Hour)
+
+	c := &Config{
+		ServerAddr: &net.UnixAddr{
+			Net:  "unix",
+			Name: apiHandler.sockPath,
+		},
+		SVID:          baseSVID,
+		SVIDKey:       baseSVIDKey,
+		Log:           testLogger,
+		TrustDomain:   url.URL{Host: trustDomain},
+		SVIDCachePath: path.Join(dir, "svid.der"),
+		Bundle:        apiHandler.bundle,
+		Tel:           &telemetry.Blackhole{},
+	}
+	mgr, err := New(c)
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	m := mgr.(*manager)
+	err = m.Start()
+	if err != nil {
+		t.Error(err)
+		return
+	}
+	defer m.Shutdown()
+
+	cert, key := m.getBaseSVIDEntry()
```

Comment:
Currently the mock api handler is doing tls (or at least that was my intention :slightly_smiling_face:). So taking another look to the tests, I think I could avoid using these validations because, as you suggest, they are already made indirectly. The only case I think I will not be able to avoid using this funciton is on the TestSVIDRotation test, when I use it to detect that an SVID rotation happened.